error messages in HTML pages updates (#707)
arXiv/arxiv-docs: source/about/accessibile_html_error_messages.md

# Error messages in HTML papers

To produce accessible HTML papers we use a LaTeXML converter created and maintained by National Library of Standards and Technology (NIST).  
{ .intro }

> The errors that you have seen when viewing papers in HTML after submission or on arXiv.org are caused when the LaTeXML converter is unable to translate certain TeX and LaTeX software packages. TeX and LaTeX, pronounced ‘tek’ and ‘LAH-tek’, are open source software packages used to accurately display complex mathematical formulae, symbols and tables in research papers. Due to the open source nature of TeX and LaTeX and its popularity, there are many packages available to choose from. This makes supporting all of them very difficult. NIST is routinely assessing and adding LaTeX packages in an effort to provide greater versatility to the LaTeXML converter.

Reviewer: ```suggestion
The errors that you have seen when viewing papers in HTML after submission or on arXiv.org are caused when the LaTeXML converter is unable to translate certain TeX and LaTeX software packages. TeX and LaTeX, pronounced ‘tek’ and ‘LAH-tek’, are open source software packages used to accurately display complex mathematical formulae, symbols and tables in research papers. Due to the open source nature of TeX and LaTeX and its popularity, there are many packages available to paper authors, and new ones being added often. This makes supporting all of them very difficult. NIST is routinely assessing and adding LaTeX packages in an effort to provide greater versatility to the LaTeXML converter.
```

## HTML errors you may see:

When browsing and reading HTML papers on arXiv.org, you may see formatting errors in red like below: 

> ![HTML LaTeX processing error for online id](images/reader-error-01.png){.mkd-img-border}

Reviewer: This an the next two images need alt text

Author: I have added alt text.

![HTML LaTeX processing error for tik Z library](images/reader-error-02.png){.mkd-img-border}

![HTML LaTeX processing error for system name and revision](images/reader-error-03.png){.mkd-img-border}

Please see an example of a HTML processing error below:

> ![html processing error](images/author_submssn_error.png){.mkd-img-full alt='Error message in a red box stating, This paper uses packages, listed below, that do not yet convert to HTML. These issues are known and are being worked on. View the list of unsupported packages.'}

Reviewer: this is great alt text!

## Frequently Asked Questions

### What is TeX?
TeX is a text formatting and markup language first released in late 1970s by [D. Knuth](https://en.wikipedia.org/wiki/TeX#Hyphenation_and_justification). It was created to properly configure and visualize mathematical formulae, tables and symbols in printed documents such as journals and books. 

### What is LaTeX?
LaTeX is a TeX macro package developed in the early 1980s by [Leslie Lamport](https://en.wikipedia.org/wiki/LaTeX#cite_note-Lamport1986-4). It provides the ability to structure a document to meet different layout requirements such as section headers, footnotes, graphics and bibliographies.

### What is TeXML? 

> TeXML is an open source process that describes the markup elements of TeX using XML syntax. This allows XML data to be presented in PDF.

Reviewer: do we use this tool? I am not aware of arXiv calling this tool, only latexml. Or does LaTeXML use this tool?

Author: I have updated the wording here and you will see it when I update this PR.

### What is LaTeXML?

> LaTeXML is an open source software package that converts LaTeX documents to XML, HTML, EPUB, JATS and TEI. 

Reviewer: Can we add here that it is authored by the NIST team of Bruce Miller and Deyan Ginev?

### Why are TeX and LaTeX so popular? 
They are beloved by academic communities for their ability to easily display complex mathematical formulae, symbols, charts and graphs that common word processing programs have difficulty with. 

TeX and LaTeX are open source software programs  that  are platform independent which allows for the ability to view documents regardless of the operating system or software set-up making collaboration simple. These programs are maintained and iterated upon by the communities that use them, and as a result, there are many packages available offering a myriad of features to format research papers. 

### Why are TeX and LaTeX so difficult to support?

> Due to the robust open source community supporting and developing TeX and LaTeX packages, there are many packages in use. NIST is continuously reviewing which features and packages are most popular to add to the LaTeXML converter in an effort to provide error free, accessible HTML papers.

Reviewer: ```suggestion
Due to the robust open source community supporting and developing TeX and LaTeX packages, there are many packages in use. NIST is continuously reviewing which features and packages are most popular to add to the LaTeXML converter in an effort to provide high fidelity, accessible conversions to HTML.
```

### Which TeX/LaTeX packages are supported?
- Please see our list of supported packages
- Please see our list of [unsupported packages](https://github.com/brucemiller/LaTeXML/wiki/Porting-LaTeX-packages-for-LaTeXML)

### Is TikZ supported?

> Sort of. See What are 'partially' supported packages?

Reviewer: ```suggestion
Partially. See the next section below.
```

### What are ‘partially’ supported packages? 

> A partially supported package is a package that has select features NIST has chosen to support in LaTeXML. For example, TikZ offers the ability to … (need to add list of features) and LaTeXML supports …. (add list of features supported) However, TikZ also provides the ability to… which LaTeXML does not support.

Reviewer: ```suggestion
A partially supported package has select features NIST is supporting now in LaTeXML, but other features that are not yet supported. For example, TikZ offers the ability to … (need to add list of features) and LaTeXML supports …. (add list of features supported) However, TikZ also provides the ability to… which LaTeXML does not support.
```

> ### What is NIST?

Reviewer: Do we need this section? There is a link to NIST in an earlier section. That seems like enough.

Reviewer: One more recommendation: a section for authors about how they can help reduce HTML errors and make papers more accessible by using packages that LaTeXML fully supports.

# Error messages in HTML papers

> HTML papers on arXiv.org will sometimes display errors. To produce accessible HTML papers we use a LaTeXML converter created and maintained by National Library of Standards and Technology (NIST). As we work to improve accessibility we share with you the causes of these errors and what authors can do to help minimize them.

Reviewer: ```suggestion
HTML papers on arXiv.org are a work in progress and will sometimes display errors. As we work to improve accessibility we share with you the causes of these errors and what authors can do to help minimize them.
```

{ .intro }

> The errors that you have seen when viewing papers in HTML after submission or on arXiv.org are caused when the LaTeXML converter is unable to translate certain TeX and LaTeX software packages. TeX and LaTeX, pronounced ‘tek’ and ‘LAH-tek’, are open source software packages used to accurately display complex mathematical formulae, symbols and tables in research papers. Due to the open source nature of TeX and LaTeX and its popularity, there are many packages available to choose from. This makes supporting all of them very difficult. NIST is routinely assessing and adding LaTeX packages in an effort to provide greater versatility to the LaTeXML converter.

Reviewer: ```suggestion
To produce accessible HTML papers we use a converter created and maintained by the LaTeXML team at the National Library of Standards and Technology (NIST). 

The errors that you have seen when viewing papers in HTML after submission or on arXiv.org are caused when the LaTeXML converter is unable to translate certain TeX and LaTeX software packages. TeX and LaTeX, pronounced ‘tek’ and ‘LAH-tek’, are open source software packages used to accurately display complex mathematical formulae, symbols and tables in research papers. Due to the open source nature of TeX and LaTeX and its popularity, there are many packages available to choose from. This makes supporting all of them very difficult. NIST is routinely assessing and adding LaTeX packages in an effort to provide greater versatility to the LaTeXML converter.
```

![html processing error](images/author_submssn_error.png){.mkd-img-full alt='Error message in a red box stating, This paper uses packages, listed below, that do not yet convert to HTML. These issues are known and are being worked on. View the list of unsupported packages.'}

![HTML LaTeX processing error for online id](images/reader-error-01.png){.mkd-img-border alt='Errors are displayed in red markup interpersed with the text. Errors are: forward slash online ID, foward slash v.g.t.c. category, forward slash author footer, foward slash system name, foward slash system name, foward slasch author orcid.'}

![HTML LaTeX processing error for tik Z library](images/reader-error-02.png){.mkd-img-border alt='Tik Z errors are displayed in red markup. Errors are: foward slash use tik z library, foward slash p.g.f. declare decorations simple line initial'}

![HTML LaTeX processing error for system name and revision](images/reader-error-03.png){.mkd-img-border alt='System name errors are displayed in red markup. Errors are: forward slash system name, foward slash revision'
}

## Steps authors can take to minimize HTML errors

> Authors and proxies who submit papers to arXiv can help minimize HTML errors by formatting submissions using one of the LateX packages fully supported by LaTeXML. [Please view the list of fully supported LaTeX packages](https://corpora.mathweb.org/corpus/arxmliv/tex_to_html/info/loaded_file).

Reviewer: ```suggestion
Authors who submit papers to arXiv can help minimize HTML errors by using LateX packages that are fully supported by LaTeXML. [Please view the list of fully supported LaTeX packages](https://corpora.mathweb.org/corpus/arxmliv/tex_to_html/info/loaded_file).
```

### What is LaTeX?
LaTeX is a software system for document layout and preparation developed in the early 1980s by [Leslie Lamport](https://en.wikipedia.org/wiki/LaTeX#cite_note-Lamport1986-4). LaTeX is designed to provide a descriptive markup language that harnesses the power of TeX in a more user-friendly way. It consists of a collection of TeX macros and a program to process LaTeX documents providing authors with convenient commands to format different layout requirements such as section headers, footnotes, graphics and bibliographies.

### What is TeXML? 
TeXML is an open source XML vocabulary for TeX. Since TeX was designed for typesetting tangible books and journals, it was necessary to create a processor to transfrom TeXML markup into TeX markup in order to utilize LaTeX defined macros to publish XML data having customizable layout options as digital portable document formats aka PDFs.

### What is LaTeXML?

> LaTeXML was developed at the [Digital Library of Mathematical Functions at NIST](https://math.nist.gov/) (National Institute of Standards and Technology) and is authored by Bruce Miller and Deyan Ginev. It's primary output is XML and a postprocessor can convert these XML documents into HTML with mathematical forumulas as images or XHTML, HTML5, EPUB with formulas as MathML. 

Reviewer: ```suggestion
LaTeXML was developed at the [Digital Library of Mathematical Functions at NIST](https://math.nist.gov/) (National Institute of Standards and Technology) and is authored by Bruce Miller and Deyan Ginev. It's primary output is XML and a postprocessor can convert these XML documents into HTML with mathematical formulas as MathML. 
```

### What is MathML?
MathML is mathematical markup language and is an XML based application. MathML is used to describe and encode mathematical notation and scientific content for distribution on the web and other applications such as scientific software, print typesetting and voice synthesis. 

### Why are TeX and LaTeX so popular? 
They are beloved by academic communities for their ability to easily display complex mathematical formulae, symbols, charts and graphs that common word processing programs have difficulty rendering.

TeX and LaTeX are open source software programs  that  are platform independent which allows for the ability to view documents regardless of the operating system or software set-up making collaboration simple. These programs are maintained and iterated upon by the communities that use them, and as a result, there are many packages available offering a myriad of features to format research papers. 

### Why are TeX and LaTeX so difficult to support?
In progress...

### Which TeX/LaTeX packages are supported?

> - Please see our list of [supported packages](https://corpora.mathweb.org/corpus/arxmliv/tex_to_html/info/loaded_file)

Reviewer: ```suggestion
- Please see this list of [supported packages](https://corpora.mathweb.org/corpus/arxmliv/tex_to_html/info/loaded_file)
```

- Please see our list of [unsupported packages](https://github.com/brucemiller/LaTeXML/wiki/Porting-LaTeX-packages-for-LaTeXML)

### Is TikZ supported?
Partially, please see 'What are partially supported packages?' below.

### What are ‘partially’ supported packages? 

> A partially supported package is a package that has select features NIST has chosen to support in LaTeXML. For example, TikZ offers the ability to … (need to add list of features) and LaTeXML supports …. (add list of features supported) However, TikZ also provides the ability to… which LaTeXML does not support.

Reviewer: ```suggestion
A partially supported package has select features NIST has chosen to support in LaTeXML. For example, TikZ offers the ability to … (need to add list of features) and LaTeXML supports …. (add list of features supported) However, TikZ also provides the ability to… which LaTeXML does not support.
```

# Error messages in HTML papers

HTML papers on arXiv.org are a work in progress and will sometimes display errors. As we work to improve accessibility we share with you the causes of these errors and what authors can do to help minimize them.
{ .intro }

To produce accessible HTML papers we use a converter created and maintained by the LaTeXML team at the National Library of Standards and Technology (NIST). 

> The errors that you have seen when viewing papers in HTML after submission or on arXiv.org are caused when the LaTeXML converter is unable to translate certain TeX and LaTeX software packages. TeX and LaTeX, pronounced ‘tek’ and ‘LAH-tek’, are open source software packages used to accurately display complex mathematical formulae, symbols and tables in research papers. Due to the open source nature of TeX and LaTeX and its popularity, there are many packages available to choose from. This makes supporting all of them very difficult. NIST is routinely assessing and adding LaTeX packages in an effort to provide greater versatility to the LaTeXML converter.

Author: I can update the pronunciation and follow what is suggested on this page: [introduction to LaTeX](https://www.latex-project.org/about/#:~:text=LaTeX%2C%20which%20is%20pronounced%20%C2%ABLah,is%20not%20a%20word%20processor!) or leave out pronunciation all together.

### What is MathML?
MathML is mathematical markup language and is an XML based application. MathML is used to describe and encode mathematical notation and scientific content for distribution on the web and other applications such as scientific software, print typesetting and voice synthesis. 

### Why are TeX and LaTeX so popular? 
They are beloved by academic communities for their ability to easily display complex mathematical formulae, symbols, charts and graphs that common word processing programs have difficulty rendering.

TeX and LaTeX are open source software programs  that  are platform independent which allows for the ability to view documents regardless of the operating system or software set-up making collaboration simple. These programs are maintained and iterated upon by the communities that use them, and as a result, there are many packages available offering a myriad of features to format research papers. 

> ### Why are TeX and LaTeX so difficult to support?

Reviewer 2: Why are we asking a question that suggests arXiv's whole strategy on this was a bad idea?

Author: At some point I recalled a discussion about why is it so challenging to support TeX and LaTeX. This was by no means meant to imply this strategy was a bad idea. I thought it more as managing expectations and indicating not all packages will be supported.
